components/kubeflow-pipeline/pipeline_components/analytics-server/flwr_analytics_server/fedbox.py

```diff
@@ -0,0 +1,101 @@
+import io
+import json
+from argparse import ArgumentParser, Namespace
+from typing import Dict, List, Tuple
+from .fivenum import compute_fivesum, FiveNum
+from .fedhist import aggregate_histograms, HistogramData
+import matplotlib.pyplot as plt
+import numpy as np
+from flwr.common import Properties, Scalar
+from .provider import AnalyticsProvider, scalar_to_numpy
+
+
+class BoxPlotProvider(AnalyticsProvider):
+    def __init__(self) -> None:
+        self._client_data: List[HistogramData] = []
+        self.nbins = 0
+        self.hrange = (0, 0)
+
+    def client_input_data(self) -> Dict[str, Scalar]:
+        return {
+            "nbins": self.nbins,
+            "hmin": self.hrange[0],
+            "hmax": self.hrange[1],
+        }
+
+    def add_client_data(self, properties: Properties) -> None:
+        self._client_data.append(
+            HistogramData(
+                counts=scalar_to_numpy(properties["counts"]),
+                bins=scalar_to_numpy(properties["bins"]),
+            )
+        )
+
+    def aggregate(self) -> None:
+        client_input = self.client_input_data()
+        hmin = client_input['hmin']
+        hmax = client_input['hmax']
+        histogram=aggregate_histograms(self._client_data)
+        fn = compute_fivesum(hmin, hmax, histogram)
+        self._result = fn
```

Comment:
Following the `HistogramProvider` inheritance consideration. This method could look like:

```suggestion
    def aggregate(self) -> None:
        super().aggregate()
        histogram = self._result

        client_input = self.client_input_data()
        self._result = compute_fivesum(client_input['hmin'], client_input['hmax'], histogram)
```

Reply:
Makes sense. Updated.